website/pizzas/static/pizzas/js/admin.js

```diff
@@ -1,112 +1,60 @@
             return;
         }
 
-        if ($(this).val() === none) {
-            $(this).removeClass('paid');
-        } else {
-            $(this).addClass('paid');
-        }
-
-        $.ajax({
-            url: '/api/v1/pizzas/orders/' + id + '/',
-            type: 'PATCH',
-            data: JSON.stringify({payment: $(this).val()}),
-            headers : {
-                'X-CSRFToken': csrfToken,
-                'Accept': 'application/json',
-                'Content-Type': 'application/json'
+        var successCallback = function(data) {
+            if (!data) {
+                select.removeClass('paid');
+            } else {
+                select.addClass('paid');
             }
-        }).done(function(data) {
-            if (data.payment === none) {
+            select.val(data ? data.payment.type: none);
+        };
+
+        var failCallback = function(xhr) {
+            select.val(payment_previous);
+
+            if (payment_previous === none) {
                 select.removeClass('paid');
             } else {
                 select.addClass('paid');
             }
-            select.val(data.payment);
-            $('table').trigger('update');
-        }).fail(function(xhr) {
+
             var data = $.parseJSON(xhr.responseText);
-            if (data.message !== undefined) {
-                alert(data.message);
-            } else if (data.payment !== undefined) {
-                alert(data.payment.join('\n'));
+            if (data.detail !== undefined) {
+                alert(data.detail);
             }
-        });
-    });
-
-    $('a.deletelink').click(function() {
-        if (confirm(gettext('Are you sure you want to delete this order?'))) {
-            var id = $(this).parents('tr').data('id');
-            var button = $(this);
+        };
 
-            $.ajax({
-                url: '/api/v1/pizzas/orders/' + id + '/',
-                type: 'DELETE',
-                headers : {
-                    'X-CSRFToken': csrfToken,
-                    'Accept': 'application/json',
-                    'Content-Type': 'application/json'
-                }
-            }).done(function() {
-                button.parents('tr').remove();
-                $('table').trigger('update');
-            }).fail(function(xhr) {
-                var data = $.parseJSON(xhr.responseText);
-                if (data.message !== undefined) {
-                    alert(data.message);
-                } else if (data.payment !== undefined) {
-                    alert(data.payment.join('\n'));
-                }
-            });
+        if ($(this).val() === none) {
+            $(this).removeClass('paid');
+            request('DELETE', url, null, successCallback, failCallback);
+        } else {
+            $(this).addClass('paid');
+            request('PATCH', url, { payment_type: $(this).val() }, successCallback, failCallback);
         }
-    });
 
-    $('#searchbar').on('input', function() {
```

Comment:
Where did this go?

Reply:
woops, went a bit too hard on the removing code

Reply:
Yeah and now added too much